cpp/demo/hyperelasticity/main.cpp

```diff
@@ -194,9 +196,36 @@ int main(int argc, char* argv[])
     la::petsc::Vector _u(la::petsc::create_vector_wrap(*u->x()), false);
     newton_solver.solve(_u.vec());
 
+    // Calculate Cauchy stress
+    // Construct appropriate Basix element for stress
+    const auto family = basix::element::family::P;
+    const auto cell_type = basix::cell::type::tetrahedron;
+    const auto variant = basix::element::lagrange_variant::equispaced;
+    const int k = 0;
+    const bool discontinuous = true;
+
+    const auto S_element
```

Comment:
I'd suggest
```
basix::FiniteElement  S_element = basix::create_element(basix::element::family::P, basix::cell::type::tetrahedron, 0, basix::element::lagrange_variant::equispaced, true);
```
Use `auto` can hinder readability.

Might be nice to get the Basix element type from the mesh. We have `mesh::cell_type_to_basix_type` for this.